docs/howtos/early_stopping.rst

```diff
@@ -0,0 +1,124 @@
+Early Stopping
+=============================
+Early stopping is used while building model to stop training loop when specific conditions are met. Some such conditions are:
+
+* Train loss not decreasing at all.
+* Train loss is decreasing but the decrease in loss is is insignificant.
+
+How to use
+-----------------------------
+
+Make the below changes in annotated  `MNIST example <https://colab.research.google.com/github/google/flax/blob/main/docs/notebooks/annotated_mnist.ipynb>`_. to achieve early stopping.
+
+.. testsetup::
+
+  # Required (hidden) code for the templates in documnetation to work
+  import jax
+
+  rng = jax.random.PRNGKey(0)
+  rng, init_rng = jax.random.split(rng)
+  num_epochs = 0
+
+  def train_epoch(state, train_ds, batch_size, epoch, input_rng, es):
+    """ this is dummy train function """
+    if es:
+      return state, es
+    return state
+
+
+Use :code:`early_stopping.EarlyStopping` class imported from :code:`flax.training` to create an early stopping object :code:`es`.
+
+:code:`min_delta` is the minimum expected decrease in loss metric. 
+:code:`es.should_stop` becomes true if decrease in the loss metric is greater than :code:`min_delta` for consecutive :code:`patience` number of epochs. 
+Otherwise, it is :code:`false` indicating that we must continue training.
+
+
+A sample training loop looks like this
```

Comment:
Can you use syntax highlighting here for the codediff blocks like in the enselmbing HOWTO? https://github.com/google/flax/blob/main/docs/howtos/ensembling.rst

You have to append "#!" to each line you want to highlight